### PR TITLE
Make tests and get the analysis to pass them

### PR DESCRIPTION
**Pull request by outofculture**

### CODE REVIEW

File: neuroanalysis/fitting/exp.py [hunk 1]
```diff
@@ -24,8 +24,8 @@ def estimate_exp_params(data):
     params : tuple
         (yoffset, yscale, tau, toffset)
     """
-    start_y = data.data[:len(data.data)//100].mean()
-    end_y = data.data[-len(data.data)//10:].mean()
+    start_y = data.data[:len(data.data)//10].mean()
+    end_y = data.data[-len(data.data)//100:].mean()
```
Comment (campagnola) [inline neuroanalysis/fitting/exp.py]: Why these?

File: neuroanalysis/fitting/exp.py [hunk 2]
```diff
@@ -42,64 +42,103 @@ def normalized_rmse(data, params, fn: Callable=exp_decay):
...
+    exp_like = 1 / (1 + err / linear_err)
+    exp_like = max(0, exp_like - 0.5) * 2
+
+    # pv_factor = 1
+    # el_factor = 4
+    # return ((poor_variation ** pv_factor) * (exp_like ** el_factor)) ** (1 / (pv_factor + el_factor))
+    return exp_like
+
+
+def exact_fit_exp(data: TSeries):
```
Comment (campagnola) [inline neuroanalysis/fitting/exp.py]: Rename to exp_fit

File: neuroanalysis/fitting/exp.py (hunk 1 above)
Comment (outofculture) [reply neuroanalysis/fitting/exp.py]: the changes at the beginning of the expected curve are more drastic than the changes at the end. thus, the last hundred samples are likely going to be the same, while the first hundred samples are not.

ooooh, wait, I just read the math. it's not count of samples, it's "tenth of all samples" v. "hundredth of all samples". I read it backwards.

File: neuroanalysis/test_pulse.py [hunk 3]
```diff
@@ -25,27 +32,72 @@ def __init__(self, rec: PatchClampRecording, indices=None):
...
         self._analysis = None
         # expose these for display and debugging
+        self._main_fit_region = None
         self.main_fit_result = None
         self.main_fit_trace = None
         self.fit_result_with_transient = None
         self.fit_trace_with_transient = None
-        self.initial_double_fit_trace = None
+
+    def save(self):
```
Comment (campagnola) [inline neuroanalysis/test_pulse.py:50]: Seems like this wil make it awkward to add a Recording.save later on.. Could this done more cleanly by starting there instead? And also adding TSeries.save/load?

File: neuroanalysis/tests/bath-ugly.h5
Comment (campagnola) [inline neuroanalysis/tests/bath-ugly.h5:1]: There is a separaate test data repo

**Pull request merged**